### PR TITLE
chore(core): remove deprecated mobilehub config parser class and rename to parseAWSExports

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

Remove deprecated mobilehub config parser class and use functional parser instead. The functional that used to be called  `parseMobileHubConfig()` is renamed to `parseAWSExports()` to reduce confusion.

#### Description of how you validated changes
Unit test & Integration test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by awsluja — APPROVED** [review]

File: packages/analytics/__tests__/Analytics.test.ts [hunk 1]
```diff
@@ -79,7 +74,7 @@ describe('Analytics test', () => {
 			analytics.configure({ mock: 'value' });
 
 			await analytics.startSession();
-			expect(hubSpy).toBeCalledWith(
+			expect(Hub.dispatch as jest.Mock).toBeCalledWith(
```
Comment (stocaaro) [inline packages/analytics/__tests__/Analytics.test.ts:77]: Can you just cast anything as a jest Mock?  This seems magical.

**Review by stocaaro — APPROVED** [review]

File: packages/storage/src/Storage.ts [hunk 2]
```diff
@@ -147,32 +151,33 @@ export class Storage {
 			'SSEKMSKeyId',
 		];
 
-		const isInStorageArrayKeys = (k: string) =>
-			storageArrayKeys.some(x => x === k);
-		const checkConfigKeysFromArray = (k: string[]) =>
-			k.find(k => isInStorageArrayKeys(k));
+		const hasDefaultProviderConfigKeys = (config: object) =>
```
Comment (cshfang) [inline packages/storage/src/Storage.ts:154]: This is much clearer than before but do you know if the behavior is correct? As it is (and was), it just checks if the config contains *any* key from the `defaultProviderConfigKeys` array ... do you know if that was the intent?

File: packages/storage/src/Storage.ts [hunk 3]
```diff
@@ -147,32 +151,33 @@ export class Storage {
...
 		) {
-			amplifyConfig.Storage[DEFAULT_PROVIDER] = {};
+			storageConfig[DEFAULT_PROVIDER] = {};
 		}
 
-		Object.entries(amplifyConfig.Storage).map(([key, value]) => {
-			if (key && isInStorageArrayKeys(key) && value !== undefined) {
-				amplifyConfig.Storage[DEFAULT_PROVIDER][key] = value;
-				delete amplifyConfig.Storage[key];
+		Object.entries(storageConfig).map(([key, value]) => {
```
Comment (cshfang) [inline packages/storage/src/Storage.ts:164]: I know this isn't your code but should this be using `map`?

File: packages/core/src/Parser.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { ConsoleLogger as Logger } from './Logger';
```
Comment (cshfang) [inline packages/core/src/Parser.ts:3]: Should the name of this file still be Parser?

**Review by cshfang — COMMENTED** [review]
Thank you so much. Looks so much better

File: packages/storage/src/Storage.ts (hunk 2 above)
Comment (AllanZhengYP) [reply packages/storage/src/Storage.ts:154]: Yes. What it does is recognize the `defaultPrivderConfigKeys` and values and remove them from the 1st layer of the config object but move to `config[provider]`. This part of code can be definitely further simplefied. I will followup on this.

File: packages/core/src/Parser.ts (hunk 4 above)
Comment (AllanZhengYP) [reply packages/core/src/Parser.ts:3]: Good callout btw. I will followup on it.

**Pull request merged**